### ISSUE TITLE
Support customizing data and time into ISO format

### ISSUE THREAD

**Issue by wenlzhang** [uphy/obsidian-reminder#8]
I see that data and time format customization is on the road map. Just want to provide some input regarding this.

It would be great to support ISO 8601 format `YYYY-MM[-DDTHH:mm:ss.nnn+ZZ]`. An example of this would be `2021-04-18T04:19:35.000`. This format is also used by another powerful plugin Dataview:
[Data Annotation - Dataview](https://blacksmithgu.github.io/obsidian-dataview/data-annotation/)

Therefore, supporting this would provide better integration among plugins.

**Comment by uphy** [comment]
Added feature to change date time format.
https://github.com/uphy/obsidian-reminder/tree/1.0.5#date-format

**Comment by uphy** [comment]
Without using the above command, you can change format with settings
<img width="1325" alt="image" src="https://user-images.githubusercontent.com/1233373/131495570-cef20eff-25ba-4e6e-b1c1-14351865c243.png">

**Comment by wenlzhang** [comment]
That's great news!

I ran a quick test by using the command. However, it did not change the format of existing tasks from `@2021-08-31 12:00` to `@2021-08-31T12:00`. It did change the format in the configuration pane. Is this the desired behaviour?

Another observation is that defining the reminder in the format of `@2021-08-31T12:00` did not give me a reminder for the specified task. Does one need to specify the full format like `2021-08-31T12:00:00:000`? It would be great if things at the end can be optional, meaning that `@2021-08-31T12:00`, `2021-08-31T12:00:00` and `2021-08-31T12:00:00:000` would all work.

**Comment by uphy** [comment]
Oh, it worked for me...

![Kapture 2021-09-03 at 21 36 50](https://user-images.githubusercontent.com/1233373/132006359-4ebb2247-8c48-4c0d-a2eb-3d41b3c89ab7.gif)

Can you try `Scan reminders`?
<img width="698" alt="image" src="https://user-images.githubusercontent.com/1233373/132006560-3c0dcdea-d80e-420e-a748-53580cf85f6c.png">

In the latest release, there is a bug not to re-scan on setting update. (it will also be fixed in the next release)

**Comment by wenlzhang** [comment]
There is no response on my side:
![CleanShot 2021-09-03 at 15 17 25](https://user-images.githubusercontent.com/44930227/132011926-fa07ed1e-f425-47f6-8cd5-896627579a6f.gif)

Also, if the date and time format is set as `YYYY-MM-DDTHH:mm:ss:SSS`, the following setting would not give me a reminder:
`- [ ] Task (@2021-09-03T12:00) #ToDoTask #CategoryResearch 📅 2021-09-30`
This is also mentioned in my previous comment.

**Comment by uphy** [comment]
The command `Convert reminder time format` converts from current format to the selected format.
So you need to set the following setting to `YYYY-MM-DD HH:mm` before you execute convert command.
<img width="970" alt="image" src="https://user-images.githubusercontent.com/1233373/132111507-183c2b08-a958-460c-8470-2622e4f0ebca.png">

> This is also mentioned in my previous comment.

Sorry but it is expected behavior.
This plugin uses 2 formats for date time.
- date and time
- date only

You cannot use different format for date and time format currently.
Why you need multiple representation for time?

**Comment by wenlzhang** [comment]
> So you need to set the following setting to `YYYY-MM-DD HH:mm` before you execute convert command.

Thanks for the clarification! It works now.

After converting the date and time format, I notice that the modification time of all Markdown notes in my vault has been updated to the time when the conversion command was executed, even for notes that do not have date/time info of any format. Can this be prevented?

- Maybe the modification time should only be updated for notes that contain data/time of the format that matches either of the three options, i.e. the Reminder plugin format, Tasks plugin format and the Kanban plugin format.
- For notes that have the time info, e.g. `2021-09-05`, but not `(@2021-09-05)`, `📅 2021-09-05` or `@{2021-09-05}`, maybe the date and time info should not be updated, i.e. the modification time of the note should not change?

> Why you need multiple representation for time?

I was thinking that the time format of `YYYY-MM-DDTHH:mm:ss:SSS` is the full format for reminder time the user specifies in the configuration pane. However, sometimes the user may use a shorthand format, e.g. `2021-09-05T12:00`, to define a reminder in the notes.

- Therefore, it might be helpful the latter format would also give a reminder, although the user set the former format in the configuration.
- This is also why I suggested in another issue that maybe two data and time formats can be defined, one for identifying tasks and the other for completing tasks.

Of course, this is not a must-have feature, since might complicate this plugin. 😅
